Sources/NIOHTTP2/ConnectionStateMachine/FrameSendingStates/SendingPushPromiseState.swift

```diff
@@ -0,0 +1,79 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+import NIOHTTP1
+
+/// A protocol that provides implementation for sending PUSH_PROMISE frames, for those states that
+/// can validly send pushed streams.
+///
+/// This protocol should only be conformed to by states for the HTTP/2 connection state machine.
+protocol SendingPushPromiseState {
+    var streamState: ConnectionStreamState { get set }
+
+    var localInitialWindowSize: UInt32 { get }
+
+    var mayPush: Bool { get }
+}
+
+extension SendingPushPromiseState {
+    mutating func sendPushPromise(originalStreamID: HTTP2StreamID, childStreamID: HTTP2StreamID, headers: HTTPHeaders) -> StateMachineResult {
+        return self._sendPushPromise(originalStreamID: originalStreamID, childStreamID: childStreamID, headers: headers)
+    }
+
+    fileprivate mutating func _sendPushPromise(originalStreamID: HTTP2StreamID, childStreamID: HTTP2StreamID, headers: HTTPHeaders) -> StateMachineResult {
```

Comment:
what makes this function and the one above (which calls it) different?

Reply:
Nothing, this is just a holdover from when I had an implementation in the override below. Will compress.

Reply:
Addressed.